src/contracts/pods/EigenPod.sol

```diff
@@ -786,6 +792,18 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         return _validatorPubkeyHashToInfo[pubkeyHash].status;
     }
 
+    /// @notice Returns the validatorInfo for a given validatorPubkey
+    function validatorPubkeyToInfo(bytes calldata validatorPubkey) external view returns (ValidatorInfo memory) {
+        return _validatorPubkeyHashToInfo[_calculateValidatorPubkeyHash(validatorPubkey)];
+    }
+
+    /// @notice Returns the validator status for a given validatorPubkey
+    function validatorStatus(bytes calldata validatorPubkey) external view returns (VALIDATOR_STATUS) {
+        require(validatorPubkey.length == 48, "EigenPod.validatorPubkeyHashToInfo must be a 48-byte BLS public key");
+        bytes32 validatorPubkeyHash = _calculateValidatorPubkeyHash(validatorPubkey);
```

Comment:
this is a duplicate check -- you have the same `require` statement in the `_calculateValidatorPubkeyHash` function. I think it's reasonable to delete this first check (since you also use the length check in `validatorPubkeyToInfo`, you should **not** delete the check in `_calculateValidatorPubkeyHash`)

Reply:
fixed!